tools/rosbag_storage/src/bag.cpp

```diff
@@ -646,9 +874,13 @@ void Bag::writeConnectionRecord(ConnectionInfo const* connection_info) {
     header[OP_FIELD_NAME]         = toHeaderString(&OP_CONNECTION);
     header[TOPIC_FIELD_NAME]      = connection_info->topic;
     header[CONNECTION_FIELD_NAME] = toHeaderString(&connection_info->id);
-    writeHeader(header);
-
-    writeHeader(*connection_info->header);
+    if (!aes_session_key_.empty()) {
+        writeEncryptedHeader(header);
```

Comment:
If the decision to write encrypted header vs. not encrypted header is done inside the `writeHeader(...)` member method, then you don't need a `writeEncryptedHeader(...)` method and it helps ABI compatibility.

Reply:
I wanted to encrypt the connection headers only.  If the `if` condition moves inside `writeHeader`, all the headers including the bag file header (containing encrypted session key) would be encrypted.